tests/libvmi/vmi.go

```diff
@@ -135,3 +135,35 @@ func baseVmi(name string) *v1.VirtualMachineInstance {
 	}
 	return vmi
 }
+
+// WithResourceCpu specifies the vmi cpu resource.
+func WithResourceCPU(value string) Option {
+	return func(vmi *v1.VirtualMachineInstance) {
+		if vmi.Spec.Domain.Resources.Requests == nil {
+			vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{}
+		}
+		vmi.Spec.Domain.Resources.Requests[k8sv1.ResourceCPU] = resource.MustParse(value)
+	}
+}
+
+// WithSerialBios if set the BIOS output will be transmitted over serial.
+func WithSerialBios() Option {
```

Comment:
Same as below

Reply:
I attempted to provide an approximate definition: add a public only after you have two independent use cases for it.

It is very hard to predefine what a good API looked like. In healthy projects there is a tension between maintainers who want to keep a small bugless API and multiple users with conflicting requests to extend the API. A good API arises gradually from this tension.